llvm/utils/TableGen/AIEVariableInstrItineraryEmitter.cpp

```diff
@@ -111,21 +123,28 @@ void AIEVariableInstrItineraryEmitter::run(raw_ostream &OS) {
     std::vector<Record *> AltItinary =
         R->getValueAsListOfDefs("ItineraryRegPairs");
     if (AltItinary.size()) {
-      int FirstOperandIdx = AltItinary[0]->getValueAsInt("OperandIdx");
-      for (const auto AltItinerary : AltItinary)
-        if (FirstOperandIdx != AltItinerary->getValueAsInt("OperandIdx"))
-          llvm_unreachable("All alternate itineraries for an instruction must "
-                           "have the same operand index");
-
-      std::vector<std::tuple<llvm::StringRef, llvm::StringRef, unsigned>>
+      std::vector<std::tuple<llvm::StringRef,
+                             std::vector<std::pair<llvm::StringRef, unsigned>>>>
           RegItineraryPairs;
       for (const auto AltItinerary : AltItinary) {
+        std::vector<Record *> OpRegType =
+            AltItinerary->getValueAsListOfDefs("RegTypeList");
+        std::vector<std::pair<llvm::StringRef, unsigned>> OperandRegClass;
+        std::unordered_set<unsigned> uniqueOperandIdx;
```

Comment:
I guess a find with a lambda in OperandRegClass would be fine to detect duplicates.